### DISCUSSION TITLE
Exclude Stationary Molecules from Compartments

### DISCUSSION THREAD

**Discussion by DR-99** [Open-Systems-Pharmacology/Forum#1338] [Questions & Problems] [answered]
Hello OSP Community!

I think I may be missing something very basic. I am currently trying to modify a model developed in PK-Sim in Mobi by adding additional compartments. My model has several stationary molecules in the form of transporters and enzymes, but they were of course added to be "active" only in specific sites. However, I noticed that all of these molecules are present under every organ in the spatial structure including the compartments that I added, and I can't seem to find a way to add a "not in container" tag or the equivalent. Unfortunately, it is resulting in errors only for the newly added compartments when trying to configure the simulation. 
A partial solution that I found is to add a tag to the individual parameters of the molecules, but this is so tedious and new errors emerge as soon as I get rid of the old ones. I have also tried searching for matching paths in the Molecule Start Values block, but even after extending, I cannot find my new compartments listed as a path element. So, is it possible to exclude molecules from containers in Mobi? Thanks in advance!

**Comment by PavelBal** [comment]
Molecules are excluded from compartments in the Molecules Start Values building block through the `IsPresent` property. I would advice to adjust the pattern in your new organ based on the pattern in any standard organ

![image](https://user-images.githubusercontent.com/25903040/195620885-568c8a2e-6306-449c-a4d4-be7dcfc2c0ae.png)

**Reply by DR-99** [reply]
Thanks for the quick response! I did suspect that this was related to the issue. However, I am unable to find my new compartments in the list, just the organs that were define in the original spatial structure. I have tried "extending" the list, and I also verified that my new compartments are under "Organism" in the spatial structure. Any ideas, what may be causing this?

**Reply by Yuri05** [reply]
Did you define your compartment(s) in the spatial structure as **physical**?
Only physical containers may contain molecules (s. https://docs.open-systems-pharmacology.org/working-with-mobi/mobi-documentation/model-building-components#spatial-structures )

**Reply by DR-99** [reply]
Yes, I made certain to define them as "physical compartments". To give you more information not only did I add compartments, but I also deleted some of them. However, the corresponding MSV block, still seems to recognize the deleted compartments since they are added each time I try to "extend" the list.

**Comment by PavelBal** [comment, marked as answer]
Do you maybe have a copy of the Spatial Structures-BB? Then the MSV-BB refers to the "old" BB and not the extended one.

**Reply by DR-99** [reply]
Yes! I had cloned the original MSV in order to make edits for my new BB. I just tried making a new one from scratch and now the problem is solved! Thanks so much for your insight!
